app/assets/javascripts/events.js

```diff
@@ -0,0 +1,12 @@
+$(function() {
+  // Set up our own class of events we can listen for to load some JS only on specific pages.
```

Comment:
This can be achieved by creating another pack file and only including it on the pages that we want that js served on. Do you think that's a better solution long term then using events to manage it?

Reply:
Will that work for Turbolinks transitions also?

Reply:
On second thought I think that using multiple pack files for each page is worse.  Since Rails doesn't support HTTP/2 it would just increase load times by needing to download multiple JS files.  Technically I think it does work with current turbolinks version as it should check the header section and merge any changes that it sees, but I'm not sure what the behavior is of old versions so I wouldn't quote me on that.

Unrelated to that though, is there a reason to keep this in sprockets as opposed to putting it in webpack?  Also I don't see anywhere in the code that is using this new event system, is this for future proof or am I missing something?

Reply:
Make sure to remember that idea.  Might come in handy for some other pages :+1: